packages/typegpu/src/smol/wgslGenerator.ts

```diff
@@ -273,9 +292,18 @@ export function generateExpression(
     }
 
     // Assuming that `id` is callable
-    return (idValue as unknown as (...args: unknown[]) => unknown)(
+    const fnRes = (idValue as unknown as (...args: unknown[]) => unknown)(
       ...resolvedResources,
     ) as Resource;
+
+    if (!isMarkedInternal(idValue)) {
+      throw new Error(`Function ${String(idValue)} is not marked as internal.`);
+    }
```

Comment:
Also, I feel like we should check this earlier, before calling the function.

```suggestion
    if (!isMarkedInternal(idValue)) {
      throw new Error(`Function ${String(idValue)} has not been created using TypeGPU APIs. Did you mean to wrap the function with tgpu.fn(args, return)(...) ?`);
    }
```